oxo.py

```diff
@@ -1,26 +1,49 @@
 class OxoBoard:
     def __init__(self):
-        """ The initialiser. Initialise any fields you need here. """
-        raise NotImplementedError("TODO: implement __init__")
+        self.loc = []
+        """Fills Board loc with 0"""
+        for i in range(0,3):
+            temp = []
+            for j in range(0,3):
+                temp.append(0)
+            self.loc.append(temp)
 
     def get_square(self, x, y):
-        """ Return 0, 1 or 2 depending on the contents of the specified square. """
-        raise NotImplementedError("TODO: implement get_square")
+        """Returns Value Of Square"""
+        return self.loc[x][y]
 
     def set_square(self, x, y, mark):
-        """ If the specified square is currently empty (0), fill it with mark and return True.
-            If the square is not empty, leave it as-is and return False. """
-        raise NotImplementedError("TODO: implement set_square")
+        """Sets Value Of A Square"""
+        if self.loc[x][y] == 0:
+            self.loc[x][y]=mark
+            return True
+        else:
+            return False
 
     def is_board_full(self):
-        """ If there are still empty squares on the board, return False.
-            If there are no empty squares, return True. """
-        raise NotImplementedError("TODO: implement is_board_full")
+        """Checks If The Board Is Full, Returns True If Board Is Full"""
+        for x in range(0,3):
+            for y in range(0,3):
+                if self.loc[x][y] == 0:
+                    return False
+        return True
+
 
     def get_winner(self):
-        """ If a player has three in a row, return 1 or 2 depending on which player.
-            Otherwise, return 0. """
-        raise NotImplementedError("TODO: implement get_winner")
+        """ Checks If there is a winner in the game returns 0 if no winner or 1/2 depending on winner"""
+        for x in range(0,3):  # checks vertical
+            for win in range(1,3):
+                if self.loc[x][0] == win and self.loc[x][1] == win and self.loc[x][2] == win:
+                    return win
+        for y in range(0,3):  # checks horizontal
+            for win in range(1,3):
+                if self.loc[0][y] == win and self.loc[1][y] == win and self.loc[2][y] == win:
+                    return win
+        for win in range(1, 3):  # checks diagonal
+            for i in range(0, 2):
```

Comment:
Slightly confusing that `win` is the inner loop in the other two cases but the outer loop here -- it makes no difference to correctness of course, but be consistent!